/*
 * Genome Nexus API
 *
 * This page shows how to use HTTP requests to access the Genome Nexus API. There are more high level clients available in Python, R, JavaScript, TypeScript and various other languages as well as a command line client to annotate MAF and VCF. See https://docs.genomenexus.org/api.  Aside from programmatic clients there are web based tools to annotate variants, see https://docs.genomenexus.org/tools.   We currently only provide long-term support for the '/annotation' endpoint. The other endpoints might change.
 *
 * API version: 2.0
 * Generated by: Swagger Codegen (https://github.com/swagger-api/swagger-codegen.git)
 */

package swagger

type Dbsnp struct {

	// class
	Class string `json:"_class,omitempty"`

	// allele_origin
	AlleleOrigin string `json:"alleleOrigin,omitempty"`

	// alleles
	Alleles []Alleles `json:"alleles,omitempty"`

	// alt
	Alt string `json:"alt,omitempty"`

	// chrom
	Chrom string `json:"chrom,omitempty"`

	// dbsnp_build
	DbsnpBuild int32 `json:"dbsnpBuild,omitempty"`

	// flags
	Flags []string `json:"flags,omitempty"`

	// hg19
	Hg19 *Hg19 `json:"hg19,omitempty"`

	// _license
	License string `json:"license,omitempty"`

	// ref
	Ref string `json:"ref,omitempty"`

	// rsid
	Rsid string `json:"rsid,omitempty"`

	// validated
	Validated bool `json:"validated,omitempty"`

	// var_subtype
	VarSubtype string `json:"varSubtype,omitempty"`

	// vartype
	Vartype string `json:"vartype,omitempty"`
}
