/*
 * Genome Nexus API
 *
 * This page shows how to use HTTP requests to access the Genome Nexus API. There are more high level clients available in Python, R, JavaScript, TypeScript and various other languages as well as a command line client to annotate MAF and VCF. See https://docs.genomenexus.org/api.  Aside from programmatic clients there are web based tools to annotate variants, see https://docs.genomenexus.org/tools.   We currently only provide long-term support for the '/annotation' endpoint. The other endpoints might change.
 *
 * API version: 2.0
 * Generated by: Swagger Codegen (https://github.com/swagger-api/swagger-codegen.git)
 */

package swagger

type EnsemblTranscript struct {

	UniprotId string `json:"uniprotId,omitempty"`

	// Ensembl transcript id
	TranscriptId string `json:"transcriptId"`

	// Ensembl gene id
	GeneId string `json:"geneId"`

	// Ensembl protein id
	ProteinId string `json:"proteinId"`

	// Length of protein
	ProteinLength int32 `json:"proteinLength,omitempty"`

	// Pfam domains
	PfamDomains []PfamDomainRange `json:"pfamDomains,omitempty"`

	// Hugo symbols
	HugoSymbols []string `json:"hugoSymbols,omitempty"`

	// RefSeq mRNA ID
	RefseqMrnaId string `json:"refseqMrnaId,omitempty"`

	// Consensus CDS (CCDS) ID
	CcdsId string `json:"ccdsId,omitempty"`

	// Exon information
	Exons []Exon `json:"exons,omitempty"`

	// UTR information
	Utrs []UntranslatedRegion `json:"utrs,omitempty"`
}
