/*
 * Genome Nexus API
 *
 * This page shows how to use HTTP requests to access the Genome Nexus API. There are more high level clients available in Python, R, JavaScript, TypeScript and various other languages as well as a command line client to annotate MAF and VCF. See https://docs.genomenexus.org/api.  Aside from programmatic clients there are web based tools to annotate variants, see https://docs.genomenexus.org/tools.   We currently only provide long-term support for the '/annotation' endpoint. The other endpoints might change.
 *
 * API version: 2.0
 * Generated by: Swagger Codegen (https://github.com/swagger-api/swagger-codegen.git)
 */

package swagger

type Hotspot struct {

	// Hugo gene symbol
	HugoSymbol string `json:"hugoSymbol,omitempty"`

	// Inframe mutation count
	InframeCount int32 `json:"inframeCount,omitempty"`

	// Missense mutation count
	MissenseCount int32 `json:"missenseCount,omitempty"`

	// Hotspot residue
	Residue string `json:"residue,omitempty"`

	// Splice mutation count
	SpliceCount int32 `json:"spliceCount,omitempty"`

	// Ensembl Transcript Id
	TranscriptId string `json:"transcriptId,omitempty"`

	// Truncation mutation count
	TruncatingCount int32 `json:"truncatingCount,omitempty"`

	// Tumor count
	TumorCount int32 `json:"tumorCount,omitempty"`

	// Hotspot type
	Type_ string `json:"type,omitempty"`
}
