/*
 * Genome Nexus API
 *
 * This page shows how to use HTTP requests to access the Genome Nexus API. There are more high level clients available in Python, R, JavaScript, TypeScript and various other languages as well as a command line client to annotate MAF and VCF. See https://docs.genomenexus.org/api.  Aside from programmatic clients there are web based tools to annotate variants, see https://docs.genomenexus.org/tools.   We currently only provide long-term support for the '/annotation' endpoint. The other endpoints might change.
 *
 * API version: 2.0
 * Generated by: Swagger Codegen (https://github.com/swagger-api/swagger-codegen.git)
 */

package swagger

type MyVariantInfo struct {

	// clinvar
	ClinVar *ClinVar `json:"clinVar,omitempty"`

	// cosmic
	Cosmic *Cosmic `json:"cosmic,omitempty"`

	// dbsnp
	Dbsnp *Dbsnp `json:"dbsnp,omitempty"`

	// gnomad_exome
	GnomadExome *Gnomad `json:"gnomadExome,omitempty"`

	// gnomad_genome
	GnomadGenome *Gnomad `json:"gnomadGenome,omitempty"`

	// hgvs
	Hgvs string `json:"hgvs,omitempty"`

	Mutdb *Mutdb `json:"mutdb,omitempty"`

	Query string `json:"query,omitempty"`

	// snpeff
	Snpeff *Snpeff `json:"snpeff,omitempty"`

	// variant
	Variant string `json:"variant,omitempty"`

	// vcf
	Vcf *Vcf `json:"vcf,omitempty"`

	// version
	Version int32 `json:"version,omitempty"`
}
