/*
 * Genome Nexus API
 *
 * This page shows how to use HTTP requests to access the Genome Nexus API. There are more high level clients available in Python, R, JavaScript, TypeScript and various other languages as well as a command line client to annotate MAF and VCF. See https://docs.genomenexus.org/api.  Aside from programmatic clients there are web based tools to annotate variants, see https://docs.genomenexus.org/tools.   We currently only provide long-term support for the '/annotation' endpoint. The other endpoints might change.
 *
 * API version: 2.0
 * Generated by: Swagger Codegen (https://github.com/swagger-api/swagger-codegen.git)
 */

package swagger

type GeneXref struct {

	// Database display name
	DbDisplayName string `json:"db_display_name"`

	// Database name
	Dbname string `json:"dbname"`

	// Description
	Description string `json:"description"`

	// Display id
	DisplayId string `json:"display_id"`

	EnsemblGeneId string `json:"ensemblGeneId,omitempty"`

	// Database info text
	InfoText string `json:"info_text,omitempty"`

	// Database info type
	InfoTypes string `json:"info_types,omitempty"`

	// Primary id
	PrimaryId string `json:"primary_id"`

	// Synonyms
	Synonyms []string `json:"synonyms,omitempty"`

	// Version
	Version string `json:"version"`
}
