/*
 * Genome Nexus API
 *
 * This page shows how to use HTTP requests to access the Genome Nexus API. There are more high level clients available in Python, R, JavaScript, TypeScript and various other languages as well as a command line client to annotate MAF and VCF. See https://docs.genomenexus.org/api.  Aside from programmatic clients there are web based tools to annotate variants, see https://docs.genomenexus.org/tools.   We currently only provide long-term support for the '/annotation' endpoint. The other endpoints might change.
 *
 * API version: 2.0
 * Generated by: Swagger Codegen (https://github.com/swagger-api/swagger-codegen.git)
 */

package swagger

type VariantAnnotationSummary struct {

	// Assembly name
	AssemblyName string `json:"assemblyName,omitempty"`

	// Canonical transcript id
	CanonicalTranscriptId string `json:"canonicalTranscriptId,omitempty"`

	// Genomic location
	GenomicLocation *GenomicLocation `json:"genomicLocation,omitempty"`

	// Strand (- or +)
	StrandSign string `json:"strandSign,omitempty"`

	// All transcript consequence summaries
	TranscriptConsequenceSummaries []TranscriptConsequenceSummary `json:"transcriptConsequenceSummaries"`

	// Most impactful transcript consequence of canonical transcript or if non-existent any transcript
	TranscriptConsequenceSummary *TranscriptConsequenceSummary `json:"transcriptConsequenceSummary"`

	// (Deprecated) Transcript consequence summaries (list of one when using annotation/, multiple when using annotation/summary/
	TranscriptConsequences []TranscriptConsequenceSummary `json:"transcriptConsequences"`

	// Variant key
	Variant string `json:"variant"`

	// Variant type
	VariantType string `json:"variantType,omitempty"`

	Vues *Vues `json:"vues,omitempty"`
}
